include/swift/AST/DiagnosticsSema.def

```diff
@@ -2988,8 +2988,8 @@ ERROR(invalid_objc_decl,none,
 ERROR(invalid_objc_swift_rooted_class,none,
       "only classes that inherit from NSObject can be declared @objc", ())
 ERROR(invalid_nonobjc_decl,none,
-      "only methods, initializers, properties and subscript declarations can "
-      "be declared @nonobjc", ())
+      "@nonobjc can only be used with members of classes, @objc protocols, and "
```

Comment:
Hm. `@nonobjc` can't actually be used with members of `@objc` protocols. It also doesn't quite make sense to note that extensions of classes must be "concrete"; for non-concrete extensions, `@nonobjc` is unnecessary because it's the default, not because it's disallowed.

(In theory you should also be able to put `@nonobjc` on a class that inherits from NSObject, but I don't think we added that feature.)